File: lib/src/logic/physics/PhysicalObject.swift
// Definition of Objects
class PhysicalObject {
    // Density of Objects
    func cdensity(mass: Double, volume: Double) -> Double {
        return mass / volume
    }

    // Energy of Objects
    func energy(mass: Double, velocity: Double) -> Double {
        return (1 / 2) * mass * (velocity * velocity)
    }

    // Force of Objects
    func force(mass: Double, acceleration: Double) -> Double {
        return mass * acceleration
    }

    // Pressure of Objects
    func pressure(force: Double, acceleration: Double) -> Double {
        return force / acceleration
    }

    // Weight of Objects
    func weight(mass: Double, gravitationalAcceleration: Double) -> Double {
        return mass * gravitationalAcceleration
    }
}

File: lib/src/logic/physics/physical_object.zig
// Definition of Objects

// Density of Objects
pub fn density(mass: f64, volume: f64) f64 {
    return mass / volume;
}

// Energy of Objects
pub fn energy(mass: f64, velocity: f64) f64 {
    return (1 / 2) * mass * (velocity * velocity);
}

// Force of Objects
pub fn force(mass: f64, acceleration: f64) f64 {
    return mass * acceleration;
}

// Pressure of Objects
pub fn pressure(force: f64, acceleration: f64) f64 {
    return force / acceleration;
}

// Weight of Objects
pub fn weight(mass: f64, gravitationalAcceleration: f64) f64 {
    return mass * gravitationalAcceleration;
}
